# Week 5 Assignment
The task for this week's assignment is to answer 3 questions.
1. What is the northernmost airport in the United States?
2. What is the easternmost airport in the United States?
3. On February 12th, 2013, which New York area airport had the windiest weather?

In [1]:
import pandas as pd
#set some pandas options controling output format
pd.set_option('display.notebook_repr_html',True) # output as flat text and not HTML
pd.set_option('display.max_rows',6) # this is the maximum number of rows we will display

In [15]:
# read the csv into a dataframe
airports = pd.read_csv("/Users/aliahsansumon/Documents/IS362/Week5/airports.csv")


In [16]:
#Display the columns in the dataframe, and get a peak at how it's formatted
airports.head()

faa                           name        lat        lon   alt  tz dst  \
0  04G              Lansdowne Airport  41.130472 -80.619583  1044  -5   A   
1  06A  Moton Field Municipal Airport  32.460572 -85.680028   264  -6   A   
2  06C            Schaumburg Regional  41.989341 -88.101243   801  -6   A   
3  06N                Randall Airport  41.431912 -74.391561   523  -5   A   
4  09J          Jekyll Island Airport  31.074472 -81.427778    11  -5   A   

              tzone  
0  America/New_York  
1   America/Chicago  
2   America/Chicago  
3  America/New_York  
4  America/New_York

In [17]:
# sort the list by latitude in descending order. This will display the airports with
# the highest latitude first.
airports.sort_values(['lat'], ascending=[False])

faa                                        name        lat         lon  \
417   EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230   BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110   AIN                          Wainwright Airport  70.638056 -159.994722   
708   K03                               Wainwright As  70.613378 -159.860350   
152   ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   
...   ...                                         ...        ...         ...   
930   MUE                               Waimea Kohala  20.001328 -155.668108   
1403  WKL                           Waikoloa Heliport  19.913600 -155.864000   
231   BSF                                Bradshaw Aaf  19.760056 -155.553717   
734   KOA                        Kona Intl At Keahole  19.738767 -156.045631   
679   ITO                                   Hilo Intl  19.721375 -155.048469   

       alt  tz dst              tzone  
417    149  -5   A                NaN  
230     44  -9   A  America/Anchorage  
110     41  -9   A  America/Anchorage  
708     35  -9   A  America/Anchorage  
152     96  -9   A  America/Anchorage  
...    ...  ..  ..                ...  
930   2671 -10   A   Pacific/Honolulu  
1403   109 -10   N   Pacific/Honolulu  
231   6190 -10   A   Pacific/Honolulu  
734     47 -10   N   Pacific/Honolulu  
679     38 -10   N   Pacific/Honolulu  

[1458 rows x 8 columns]

# Question 1 Answer- So based on the result above it shows that the northernmost airport 
in the U.S is EEN or Dillant
Airport, based on the inforamtion of highest latitude.
Since I had to also research, I discovered that the northernmost airport in the U.S is not Dillant airport.
The northernmsot airport is Wiley Post Will Rogers Memorial airport.

In [18]:
# sort the list by latitude in descending order. This will display the airports with
# the highest latitude first.
pd.set_option('display.max_rows',10) # this is the maximum number of rows we will display
airports.sort_values(['tz','lon'], ascending=[False, False])

faa                           name        lat         lon   alt  tz dst  \
942  MYF               Montgomery Field  32.475900  117.759000    17   8   A   
396  DVT  Deer Valley Municipal Airport  33.411700  112.457000  1478   8   A   
417  EEN        Dillant Hopkins Airport  72.270833   42.898333   149  -5   A   
444  EPM     Eastport Municipal Airport  44.910111  -67.012694    45  -5   A   
624  HUL                   Houlton Intl  46.123083  -67.792056   489  -5   A   
..   ...                            ...        ...         ...   ...  ..  ..   
600  HNL                  Honolulu Intl  21.318681 -157.922428    13 -10   N   
587  HHI                    Wheeler Aaf  21.483500 -158.039667   837 -10   A   
580  HDH                     Dillingham  21.579475 -158.197281    14 -10   A   
794  LIH                          Lihue  21.975983 -159.338958   153 -10   N   
207  BKH             Barking Sands Pmrf  22.022833 -159.785000    23 -10   A   

                tzone  
942    Asia/Chongqing  
396    Asia/Chongqing  
417               NaN  
444  America/New_York  
624  America/New_York  
..                ...  
600  Pacific/Honolulu  
587  Pacific/Honolulu  
580  Pacific/Honolulu  
794  Pacific/Honolulu  
207  Pacific/Honolulu  

[1458 rows x 8 columns]

# Question 2 Answer. Based on the result above is shows that the easternmost airport in the U.S is Montgomery Field 
airport, but based on my research I found that the easternmost airport is not Montgomery Field airport, or 
Deer Valley Municipal airport or Dillant Hopkins aiprot. My research shows that the easternmost airport in the U.S is
Eastport Municipal Airport.

In [19]:

# read the csv into a dataframe
weather = pd.read_csv("/Users/aliahsansumon/Documents/IS362/Week5/weather.csv")

In [20]:
# Display the columns in the dataframe, and get a peak at how it's formatted
weather.head()


origin  year  month  day  hour   temp   dewp  humid  wind_dir  wind_speed  \
0    EWR  2013      1    1     1  39.02  26.06  59.37     270.0    10.35702   
1    EWR  2013      1    1     2  39.02  26.96  61.63     250.0     8.05546   
2    EWR  2013      1    1     3  39.02  28.04  64.43     240.0    11.50780   
3    EWR  2013      1    1     4  39.92  28.04  62.21     250.0    12.65858   
4    EWR  2013      1    1     5  39.02  28.04  64.43     260.0    12.65858   

   wind_gust  precip  pressure  visib             time_hour  
0        NaN     0.0    1012.0   10.0  2013-01-01T06:00:00Z  
1        NaN     0.0    1012.3   10.0  2013-01-01T07:00:00Z  
2        NaN     0.0    1012.5   10.0  2013-01-01T08:00:00Z  
3        NaN     0.0    1012.2   10.0  2013-01-01T09:00:00Z  
4        NaN     0.0    1011.9   10.0  2013-01-01T10:00:00Z

In [25]:
weather[((weather.year == 2013) & \
         (weather.month == 2)) & \
        (weather.day == 12) & \
        (weather.wind_speed < 350)] \
        .sort_values('wind_speed', ascending = False)

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
18417    LGA  2013      2   12     2  42.98  26.06  50.94     290.0   
1018     EWR  2013      2   12    12  44.06  26.06  48.87     270.0   
18428    LGA  2013      2   12    13  44.06  23.00  43.02     300.0   
1008     EWR  2013      2   12     2  39.92  28.04  62.21     270.0   
18429    LGA  2013      2   12    14  44.06  23.00  43.02     300.0   
...      ...   ...    ...  ...   ...    ...    ...    ...       ...   
1006     EWR  2013      2   12     0  39.92  39.02  96.55     240.0   
1027     EWR  2013      2   12    21  35.96  24.08  61.67     300.0   
18415    LGA  2013      2   12     0  42.08  37.40  86.89       NaN   
9709     JFK  2013      2   12     0  42.08  37.04  82.17     280.0   
1029     EWR  2013      2   12    23  35.96  23.00  58.94     250.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
18417    23.01560   31.07106     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
1018     21.86482   31.07106     0.0    1012.5   10.0  2013-02-12T17:00:00Z  
18428    21.86482   25.31716     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
1008     20.71404   25.31716     0.0    1007.8   10.0  2013-02-12T07:00:00Z  
18429    20.71404   25.31716     0.0    1011.5   10.0  2013-02-12T19:00:00Z  
...           ...        ...     ...       ...    ...                   ...  
1006      6.90468        NaN     0.0    1006.9   10.0  2013-02-12T05:00:00Z  
1027      5.75390        NaN     0.0    1013.0   10.0  2013-02-13T02:00:00Z  
18415     4.60312        NaN     0.0       NaN   10.0  2013-02-12T05:00:00Z  
9709      4.60312        NaN     0.0    1007.1   10.0  2013-02-12T05:00:00Z  
1029      4.60312        NaN     0.0    1013.7   10.0  2013-02-13T04:00:00Z  

[71 rows x 15 columns]

In [26]:
# To account for the fact that the wind data is taken every hour I will take the average hourly 
# windspeed for each airport that day.
weather[((weather.year == 2013) & \
         (weather.month == 2)) & \
        (weather.day == 12) & \
        (weather.wind_speed < 350)] \
        .groupby('origin').wind_speed.mean()

origin
EWR    13.258987
JFK    14.384750
LGA    14.960140
Name: wind_speed, dtype: float64

# Question 3 Answer. Based on the information and data above and from my research I came to a conclusion that 
on February 12, 2013 Laguardia airport had the windest weather, because the windspeed for Laguardia that day shows 
14.960140 which is the highest than the other airports.